lib/src/view/edit_mode.dart

```diff
@@ -32,10 +32,14 @@ class EditModeComponent extends UiComponent2<EditModeProps> with RedrawCounterMi
       ...[for (var choice in EditModeChoice.values) this._button_for_choice(choice)],
     ]);
   }
+  static button_response(EditModeChoice mode){
+    app.dispatch(actions.EditModeToggle(mode));
+    app.dispatch(actions.SelectionsClear());
+  }
 
   ReactElement _button_for_choice(EditModeChoice mode) {
     return (Dom.button()
-      ..onClick = ((_) => app.dispatch(actions.EditModeToggle(mode)))
+      ..onClick = ((_) => button_response(mode))
```

Comment:
Can you put this inline? I prefer that when the event handler is small and only needs to be called once. Since it's more than a one-liner, the format will be 

```dart
..onClick = (_) {
  app.dispatch(actions.EditModeToggle(mode));
  app.dispatch(actions.SelectionsClear());
}
```